src/Misc/layoutroot/run.sh

```diff
@@ -9,16 +9,28 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
     [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+# Set job control
+set -m
+
+trap 'kill -INT -$PID' INT TERM
+
 # run the helper process which keep the listener alive
 while :;
 do
     cp -f "$DIR"/run-helper.sh.template "$DIR"/run-helper.sh
-    "$DIR"/run-helper.sh $*
+    "$DIR"/run-helper.sh $* &
```

Comment:
should we move the change into the `run-helper.sh` since that's. more close to the runner process?

Reply:
I guess we can't since this is the entry script?

Reply:
Yes, the run.sh will get the SIGTERM. That is why I installed signal handling here. I we move it to run-helper, then the run.sh will not wait for it and it is the entrypoint that receives signal